Mathlib/Data/Nat/Digits/Lemmas.lean

```diff
@@ -319,4 +319,216 @@ theorem getD_digits (n i : ℕ) {b : ℕ} (h : 2 ≤ b) : (digits b n).getD i 0
       simp
     · simp [IH _ (le_of_lt_succ (div_lt_self' n b)), pow_succ', Nat.div_div_eq_div_mul]
 
+/-! ### Bijection -/
+
+open List
+
+/--
+The list of digits of `n` in base `b` with some `0`'s appended so that its length is equal to `l`
+if it is `< l`. This is an inverse function of `Nat.ofDigits` for `n < b ^ l`,
+see `Nat.setInvOn_digitsAppend_ofDigits`.
+If `n ≥ b ^ l`, then the list of digits of `n` in base `b` is of length at least `l` and
+this function just return `b.digits n`.
+-/
+abbrev digitsAppend (b l n : ℕ) : List ℕ :=
```

Comment:
I think this and the other definitions in your PR should be `def`s which are explicitly unfolded in the handful of places where that is necessary. I think `abbrev` should be seen as close to a last resort, and if the function is characterized by some property in some meaningful way (which is the case for your functions, as they are (partial) inverses of other, simpler, functions), then we should encourage using those characterizations instead of having Lean unfold the function automatically.